ddtrace/settings/config.py

```diff
@@ -204,6 +205,9 @@ def __init__(self):
         self.env = os.getenv("DD_ENV") or self.tags.get("env")
         self.service = os.getenv("DD_SERVICE", default=self.tags.get("service"))
         self.version = os.getenv("DD_VERSION", default=self.tags.get("version"))
+
+        if self.version:
+            self.version = subprocess.check_output(["git", "rev-parse", "--short", "HEAD"]).decode("ascii").strip()
```

Comment:
This also assumes that `git` is available and the command will be successful, we'll need error handling here.

Reply:
What are the benefits of implementing this on our end instead of setting the `DD_VERSION` on the user-side?

Reply:
This would be better handled by setting the environment variable for start command for an instrumented application:

```
DD_VERSION=$(git rev-parse --short HEAD)
```